google/auth/aio/transport/sessions.py

```diff
@@ -0,0 +1,283 @@
+            and the external package `aiohttp` is not installed.
+        - google.auth.exceptions.InvalidType: If the provided credentials are
+            not of type `google.auth.aio.credentials.Credentials`.
+    """
+
+    def __init__(
+        self, credentials: Credentials, auth_request: transport.Request = None
+    ):
+        if not isinstance(credentials, Credentials):
+            raise exceptions.InvalidType(
+                f"The configured credentials of type {type(credentials)} are invalid and must be of type `google.auth.aio.credentials.Credentials`"
+            )
+
+        self._credentials = credentials
+
+        self._auth_request = auth_request or (AIOHTTP_INSTALLED and AiohttpRequest())
+        if not self._auth_request:
+            raise exceptions.TransportError(
+                "`auth_request` must either be configured or the external package `aiohttp` must be installed to use the default value."
+            )
+
+    async def request(
+        self,
+        method: str,
+        url: str,
+        data: bytes = None,
+        headers: Mapping[str, str] = None,
+        max_allowed_time: Optional[
+            float
+        ] = transport._DEFAULT_TIMEOUT_SECONDS,  # TODO (ohmayr): Do we want the default value for both to be the same?s
+        timeout: float = transport._DEFAULT_TIMEOUT_SECONDS,
+        **kwargs,
+    ) -> transport.Response:
+        """
+        Args:
+                method (str): The http method used to make the request.
+                url (str): The URI to be requested.
+                data (bytes): The payload or body in HTTP request.
+                headers (Mapping[str, str]): Request headers.
+                timeout (float):
+                The amount of time in seconds to wait for the server response
+                with each individual request.
+                max_allowed_time (float):
+                If the method runs longer than this, a ``Timeout`` exception is
+                automatically raised. Unlike the ``timeout`` parameter, this
+                value applies to the total method execution time, even if
+                multiple requests are made under the hood.
+
+                Mind that it is not guaranteed that the timeout error is raised
+                at ``max_allowed_time``. It might take longer, for example, if
+                an underlying request takes a lot of time, but the request
+                itself does not timeout, e.g. if a large file is being
+                transmitted. The timout error will be raised after such
+                request completes.
+
+        Returns:
+                google.auth.aio.transport.Response: The HTTP response.
+
+        Raises:
+                google.auth.exceptions.TimeoutError: If the method does not complete within
+                the configured `max_allowed_time` or the request exceeds the configured
+                `timeout`.
+
+        # TODO (ohmayr): Investigate if this is required.
```

Comment:
Is this in reference to the response headers? I don't think we should try to mutate them.

Reply:
addressed.